src/backend/distributed/utils/citus_clauses.c

```diff
@@ -114,7 +114,19 @@ PartiallyEvaluateExpression(Node *expression,
 	}
 	else if (ShouldEvaluateExpressionType(nodeTag))
 	{
-		if (FindNodeCheck(expression, IsVarNode))
+		if (masterEvaluationContext &&
+			masterEvaluationContext->evaluationMode == EvaluateParams &&
+			FindNodeCheck(expression, IsParam))
```

Comment:
This `FindNodeCheck` is still a little odd. Why don't we just keep recursing into `PartiallyEvaluateExpression` if `evaluationMode == EvaluateParams`? Eventually we'll find the Param and evaluate it.

In other words, we should have something like `if (ShouldEvaluateExpressionType(nodeTag) && masterEvaluationContext->evaluationMode == EvaluateFunctionsAndParams)` and just fall into the `else { }` block in the `EvaluateParams` case.

Reply:
makes  sense